lib/models/application.js

```diff
@@ -661,20 +629,20 @@ const getApplicationModel = function (deps, opts) {
 		 * 	if (error) throw error;
 		 * });
 		 */
-		remove: (nameOrSlugOrId) =>
-			getId(nameOrSlugOrId)
-				.then((applicationId) =>
-					pine.delete({
-						resource: 'application',
-						id: applicationId,
-					}),
-				)
-				.catch((err) => {
-					if (isNotFoundResponse(err)) {
-						treatAsMissingApplication(nameOrSlugOrId, err);
-					}
-					throw err;
-				}),
+		remove: async (nameOrSlugOrId) => {
+			try {
+				const applicationId = await getId(nameOrSlugOrId);
+				return pine.delete({
```

Comment:
```suggestion
				return await pine.delete({
```
Otherwise the catch will not run if `pine.delete` fails, which is more of an issue if `nameOrSlugOrId` is a number, in which case `getId` doesn't check with the API.